### ISSUE TITLE
[Bug] MSAL on UWP not working with Http Proxy (WAB does not take OS proxy)

### ISSUE THREAD

**Issue by JeffMushmo** [AzureAD/microsoft-authentication-library-for-dotnet#1482]
- Microsoft.Identity.Client v4.5.1 (MSAL)
- UWP (Windows 10 & Windows IoT Core)

When routing traffic through a proxy server desktop/mobile interactive authentication is having the issue of our app that is currently in production.

**Repro**
1. Install & Run a proxy server (CCProxy 8.0)
2. Configure Windows to use CCProxy as the main proxy for the OS.
3. Run application and attempt to login with the MSAL library.

```csharp
_azureAdClientApplication = 
    PublicClientApplicationBuilder.Create(appConfig.AuthScheme.ApiKey)
    .WithAuthority(appConfig.AuthScheme.Authority)
    .Build();

var interactiveBuilder = _azureAdClientApplication.AcquireTokenInteractive(new[] { appConfig.AuthScheme.Scope });
var result = await interactiveBuilder.ExecuteAsync(default(CancellationToken));
```

**Expected behavior**
When the call to ExecuteAsync is made, the user would see the first page in the Azure AD login workflow. This works when there is no proxy server.

**Actual behavior**
With a proxy configured, the auth dialog display this message: "We can't connect to the service you need right now. Check your network connection or try this again later." The only option I have is to close it.

The exception I get has the following message:
Web Authentication Broker (WAB) authentication failed. To collect WAB logs, please follow https://aka.ms/msal-net-wab-logs

Its inner exception is a FileNotFoundException.

**Comment by jennyf19** [comment]
@JeffMushmo Are you providing your own [httpClient](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/HttpClient#providing-your-own-httpclient-supporting-http-proxy-and-customization-of-user-agent-headers)?

**Comment by JeffMushmo** [comment]
@jennyf19 I've tried by using my own HttpClient and without. Was about to create a new issue that it doesn't seem to respect my HttpClientHandler/WebProxy when I supply one. Do I need to be doing both?

**Comment by jennyf19** [comment]
@JeffMushmo Can you share the settings you're using w/the WebProxy?

**Comment by JeffMushmo** [comment]
``` csharp
return new HttpClient(new HttpClientHandler
{
    UseDefaultCredentials = true,
    UseProxy = true,
    Proxy = new WebProxy("10.12.0.186", 808),
    ClientCertificateOptions = ClientCertificateOption.Automatic,
    AllowAutoRedirect = false
})
```

**Comment by jennyf19** [comment]
@JeffMushmo What is the issue you see when you use your own HttpClient?

**Comment by JeffMushmo** [comment]
It doesn't use the proxy. The requests seem to bypass it.

**Comment by jmprieur** [comment]
@JeffMushmo : did you enable enough capabilities. See https://github.com/azure-samples/active-directory-xamarin-native-v2#uwp-specific-considerations

**Comment by JeffMushmo** [comment]
<Capability Name="internetClient" />
<Capability Name="internetClientServer" />
<Capability Name="privateNetworkClientServer" />
<uap:Capability Name="sharedUserCertificates" />
<uap:Capability Name="enterpriseAuthentication" />

**Comment by bgavrilMS** [comment]
2 ideas here: 

1. Consider using this HttpClient: https://docs.microsoft.com/en-us/previous-versions/windows/apps/dn440594(v=win.10)?redirectedfrom=MSDN

2. The interactive part of the authentication, when the browser pops up, is handled by an UWP component named WAB - Windows Authentication Broker. I do not see such configuration for it, but I can contact the owning team to find out if this is the issue.

**Comment by JeffMushmo** [comment]
Thanks for the recommendations.

1. I'm not sure how this would work considering the IMsalHttpClientFactory is expecting the System.Net.Http.HttpClient type and won't build if I use the one you suggested.

2. I saw this same note for the WPF version of MSAL. If you could reach out, that would be appreciated.

**Comment by JeffMushmo** [comment]
From the perspective of implementing proxy support within the app itself, it looks like we're trying to track down where the issue might be occurring.

From the perspective of configuring my OS and allowing apps to behave naturally through the built-in handling for proxies, this still doesn't work either.

While I'm fine coding around the issue and adding the capability to my app, it would seem more urgent to fix when a proxy is configured at the OS level.

**Comment by bgavrilMS** [comment]
WAB does not seem to support injecting an HttpClient or a proxy, so the way around this would be to configure the proxy for the entire OS. 

It looks like you did this and WAB does not obey the settings? Would it be possible to get a Fiddler trace as described [here ](https://docs.microsoft.com/en-us/windows/win32/secauthn/web-authentication-problems#using-fiddler-with-web-authentication-broker)

**Comment by jmprieur** [comment]
Add limitation paragraph in UWP specific doc

**Comment by JeffMushmo** [comment]
By following the Fiddler trace instructions, I was able to get the proxy (from the OS level) working. Must have been the local loopback and private network settings. Still not able to configure proxy within runtime for WAB.

I tested this on Windows IoT Core for our application that's running as a background service. It appears the UWP runtime on IoT Core doesn't respect the proxy settings in the registry. We've confirmed this for other environments on IoT Core as well, such as PowerShell.

Unfortunately, as a result, there is no way to authenticate users with MSAL and Azure AD on Windows IoT Core if the device is sitting behind an internet proxy.

I've created an issue for the IoT Core team here:
[https://github.com/ms-iot/iot-adk-addonkit/issues/328](https://github.com/ms-iot/iot-adk-addonkit/issues/328)
